File: twoSum/twoSum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val indexMap = mutableMapOf<Int, Int>()
        nums.forEachIndexed { index, num ->
            val complement = target - num
            indexMap[complement]?.let { return intArrayOf(it, index) }
            indexMap[num] = index
        }
        throw IllegalArgumentException("No two elements sum up to the target value")
    }
}

File: twoSum/twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = {}
        # traverse the list
        for i,num in enumerate(nums):
            # check if the target-num is in the dictionary  
            if target-num in dic:
                # return the value of the dictionary along with the current index
                return [dic[target-num],i]
            dic[num] = i
        